### DISCUSSION TITLE
MQTT messages

### DISCUSSION THREAD

**Discussion by odroegehorn** [syssi/esphome-soyosource-gtn-virtual-meter#97] [General]
Hi again,
I have a question about the MQTT-messages the code is sending:

I got in my Broker:

```
soyosource_line1_c8c9a30bcac6:soyosource-line1/debug:.* debug
soyosource_line1_c8c9a30bcac6:soyosource-line1/binary_sensor/soyosource-line1_fan_running/state:.* state
soyosource_line1_c8c9a30bcac6:soyosource-line1/binary_sensor/soyosource-line1_limiter_connected/state:.* state
soyosource_line1_c8c9a30bcac6:soyosource-line1/number/soyosource-line1_start_voltage/state:.* state
soyosource_line1_c8c9a30bcac6:soyosource-line1/number/soyosource-line1_shutdown_voltage/state:.* state
soyosource_line1_c8c9a30bcac6:soyosource-line1/number/soyosource-line1_output_power_limit/state:.* state
soyosource_line1_c8c9a30bcac6:soyosource-line1/number/soyosource-line1_start_delay/state:.* state
soyosource_line1_c8c9a30bcac6:soyosource-line1/select/soyosource-line1_operation_mode/state:.* state
soyosource_line1_c8c9a30bcac6:soyosource-line1/sensor/dummy_powermeter/state:.* state
soyosource_line1_c8c9a30bcac6:soyosource-line1/sensor/soyosource-line1_battery_voltage/state:.* state
soyosource_line1_c8c9a30bcac6:soyosource-line1/sensor/soyosource-line1_battery_current/state:.* state
soyosource_line1_c8c9a30bcac6:soyosource-line1/sensor/soyosource-line1_battery_power/state:.* state
soyosource_line1_c8c9a30bcac6:soyosource-line1/sensor/soyosource-line1_ac_voltage/state:.* state
soyosource_line1_c8c9a30bcac6:soyosource-line1/sensor/soyosource-line1_ac_frequency/state:.* state
soyosource_line1_c8c9a30bcac6:soyosource-line1/sensor/soyosource-line1_temperature/state:.* state
soyosource_line1_c8c9a30bcac6:soyosource-line1/sensor/soyosource-line1_error_bitmask/state:.* state
soyosource_line1_c8c9a30bcac6:soyosource-line1/sensor/soyosource-line1_operation_mode_id/state:.* state
soyosource_line1_c8c9a30bcac6:soyosource-line1/sensor/soyosource-line1_operation_status_id/state:.* state
soyosource_line1_c8c9a30bcac6:soyosource-line1/sensor/soyosource-line1_output_power/state:.* state
soyosource_line1_c8c9a30bcac6:soyosource-line1/sensor/soyosource-line1_total_energy/state:.* state
soyosource_line1_c8c9a30bcac6:soyosource-line1/sensor/soyosource-line1_errors/state:.* state
soyosource_line1_c8c9a30bcac6:soyosource-line1/sensor/soyosource-line1_operation_mode/state:.* state
soyosource_line1_c8c9a30bcac6:soyosource-line1/sensor/soyosource-line1_operation_status/state:.* state
```

These are state-requests, I guess ... so it's send out periodically (refresh-rated) to report the state. OK?

Then I got also something with "status" ...., what's that?
```
soyosource_line1_c8c9a30bcac6:soyosource-line1/status:.* status
```

And then I got a lot of .../config messages with starting homeassistant. What's that?
In the settings I have seen, that commands are issued with:
"soyosource-line1/select/soyosource-line1_operation_mode/command"
So what are the ../config topics for?

```
soyosource_line1_c8c9a30bcac6:homeassistant/binary_sensor/soyosource-line1/soyosource-line1_fan_running/config:.* 
soyosource_line1_c8c9a30bcac6:homeassistant/binary_sensor/soyosource-line1/soyosource-line1_limiter_connected/config:.* 
soyosource_line1_c8c9a30bcac6:homeassistant/button/soyosource-line1/soyosource-line1_restart/config:.* 
soyosource_line1_c8c9a30bcac6:homeassistant/number/soyosource-line1/soyosource-line1_start_voltage/config:.* 
soyosource_line1_c8c9a30bcac6:homeassistant/number/soyosource-line1/soyosource-line1_shutdown_voltage/config:.* 
soyosource_line1_c8c9a30bcac6:homeassistant/number/soyosource-line1/soyosource-line1_output_power_limit/config:.* 
soyosource_line1_c8c9a30bcac6:homeassistant/number/soyosource-line1/soyosource-line1_start_delay/config:.* 
soyosource_line1_c8c9a30bcac6:homeassistant/select/soyosource-line1/soyosource-line1_operation_mode/config:.*
soyosource_line1_c8c9a30bcac6:homeassistant/sensor/soyosource-line1/dummy_powermeter/config:.* 
soyosource_line1_c8c9a30bcac6:homeassistant/sensor/soyosource-line1/soyosource-line1_battery_voltage/config:.* 
soyosource_line1_c8c9a30bcac6:homeassistant/sensor/soyosource-line1/soyosource-line1_battery_current/config:.* 
soyosource_line1_c8c9a30bcac6:homeassistant/sensor/soyosource-line1/soyosource-line1_battery_power/config:.* 
soyosource_line1_c8c9a30bcac6:homeassistant/sensor/soyosource-line1/soyosource-line1_ac_voltage/config:.* 
soyosource_line1_c8c9a30bcac6:homeassistant/sensor/soyosource-line1/soyosource-line1_ac_frequency/config:.* 
soyosource_line1_c8c9a30bcac6:homeassistant/sensor/soyosource-line1/soyosource-line1_temperature/config:.* 
soyosource_line1_c8c9a30bcac6:homeassistant/sensor/soyosource-line1/soyosource-line1_error_bitmask/config:.* 
soyosource_line1_c8c9a30bcac6:homeassistant/sensor/soyosource-line1/soyosource-line1_operation_mode_id/config:.* 
soyosource_line1_c8c9a30bcac6:homeassistant/sensor/soyosource-line1/soyosource-line1_operation_status_id/config:.* 
soyosource_line1_c8c9a30bcac6:homeassistant/sensor/soyosource-line1/soyosource-line1_total_energy/config:.* 
soyosource_line1_c8c9a30bcac6:homeassistant/sensor/soyosource-line1/soyosource-line1_output_power/config:.* 
soyosource_line1_c8c9a30bcac6:homeassistant/sensor/soyosource-line1/soyosource-line1_errors/config:.* 
soyosource_line1_c8c9a30bcac6:homeassistant/sensor/soyosource-line1/soyosource-line1_operation_mode/config:.* 
soyosource_line1_c8c9a30bcac6:homeassistant/sensor/soyosource-line1/soyosource-line1_operation_status/config:.* 
```

If they belong to a list of settings, that can be made, why is the device reporting them? For discovery in HA?
And if the are for config, what's the difference to .../command?
And if again, then, what are allowed parameter-values? Are they defined somewhere?

Sorry for all the questions. As I don't know HomeAssistant I need to reassemble the behaviour in FHEM.
Thanks in advance.

**Comment by syssi** [comment]
In ESPHome every entity has a `state`. The state of a temperature sensor could be `5 °C` f.e. To avoid confusion I try to use the word "measurement" if I talk about sensor states. Measurements are reported periodically depending on the `update_interval`. You could change to update interval if you like.

The `config` topics are used to help Home Automation Software (especially Home Assistant) to auto-discover the ESPHome node and register an entity per topic auto-magically. If ioBroker/FHEM doesn't support this kind of auto-discovery you could disable the feature to avoid noise:

```
mqtt:
  broker: !secret mqtt_host
  username: !secret mqtt_username
  password: !secret mqtt_password
  id: mqtt_client
  discovery: false
```

The `status` topic is explained here: https://esphome.io/components/mqtt.html#last-will-and-birth-messages
It can be used as last will testament: https://www.hivemq.com/blog/mqtt-essentials-part-9-last-will-and-testament/

Feel free to ask more questions if I missed one of your questions already. ;-)

**Reply by odroegehorn** [reply]
OK, thanks.
Seems, that I got it working.
Is there a way to suppress the "debug" message in the status frame? 
Because: in there are a lot of different things, that needs parsing ....
Thnx.

**Reply by syssi** [reply]
Could you provide an example of the payload?

**Reply by syssi** [reply]
Are you sure we are talking about the `status` topic or the `debug` topic?

**Reply by odroegehorn** [reply]
Of course:

[0;32m[I][ota:113]: Boot seems successful, resetting boot loop counter.[0m ....

or:

[0;32m[I][soyosource_display:350]:   Maximum PV voltage: 90 V[0m ...

all these are truncated, I guess due to special characters in there?

Can we suppress this debug message completely?

Thanx.

**Reply by syssi** [reply]
This can be disabled. Could you tell me the exact topic?

**Reply by syssi** [reply]
This is traffic of the `logs` topic and can be disabled. The `status` topic should report `online` or `offline`.

Is assume this is what you are looking for: https://github.com/esphome/issues/issues/3047#issuecomment-1040758180

```
mqtt:
  log_topic: null
```

**Reply by odroegehorn** [reply]
You are right. It's the "debug" topic.
"status" topic provides online/offline, correct.

With "log-topic: null" I can disable the "debug" topic?
Would be great.
Thnx.

**Reply by syssi** [reply]
Yes. The `log_topic` is the `debug` topic as far as I know. It should by mentioned on startup.

**Reply by syssi** [reply]
I recommend to decrease the log level (to `INFO` or `WARN`) instead of removing the topic.
